### PR TITLE
feat(mocks): gen all mocks through mockery

### PR DESCRIPTION
**Pull request by luphaz**
## What does this PR do?

- [ ] Adds new functionality
- [x] Alters existing functionality
- [ ] Fixes a bug
- [x] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- this PR uses `go:generate mockery` to generate all mocks
- it replace all untyped calls by `EXPECT`
- weird fixes will be highlighted as comments 

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [x] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: injector/network_disruption.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type linkOperation func([]string, string, string) error
 type networkDisruptionService struct {
 	ip       *net.IPNet
 	port     int
-	protocol network.Protocol
+	protocol v1.Protocol
```
Comment (luphaz) [inline injector/network_disruption.go:34]: keep original type so we can improve the safeness of our internal type `protocol` by making it private.

File: injector/node_failure.go [hunk 2]
```diff
@@ -50,6 +51,10 @@ func NewNodeFailureInjector(spec v1beta1.NodeFailureSpec, config NodeFailureInje
 		return nil, fmt.Errorf("environment variable %s doesn't exist", env.InjectorMountSysrqTrigger)
 	}
 
+	if config.WaitBeforeShutdown == 0 {
```
Comment (luphaz) [inline injector/node_failure.go:54]: Improve testing by avoiding waiting 10s for nothing, keeping default value to 10s like before

File: network/protocol.go [hunk 3]
```diff
@@ -5,16 +5,39 @@
...
+
+type protocolString interface {
+	string | v1.Protocol
+}
+
+// NewProtocol returns a protocol value based on the given string, possible values are:
+// - tcp: by default if value is not recognised or is tcp
+// - udp: if provided value is udp in any case
+// - arp: if provided value is arp in any case
+func NewProtocol[C protocolString](protocol C) protocol {
```
Comment (luphaz) [inline network/protocol.go:34]: We force any usage of `protocol` to use this method.
So far only 2 types are allowed to create a `protocol`, `string | v1.Protocol`

File: network/tc.go [hunk 4]
```diff
@@ -56,20 +56,24 @@ const (
...
 )
 
+//go:generate mockery --name=TrafficController --filename=tc_mock.go
+
 // TrafficController is an interface being able to interact with the host
 // queueing discipline
 type TrafficController interface {
 	AddNetem(ifaces []string, parent string, handle string, delay time.Duration, delayJitter time.Duration, drop int, corrupt int, duplicate int) error
 	AddPrio(ifaces []string, parent string, handle string, bands uint32, priomap [16]uint32) error
-	AddFilter(ifaces []string, parent string, handle string, srcIP, dstIP *net.IPNet, srcPort, dstPort int, protocol Protocol, connState connState, flowid string) (uint32, error)
```
Comment (luphaz) [inline network/tc.go:64]: mocks generation failed with parameter name being the same as private type name

File: network/tc.go [hunk 5]
```diff
@@ -56,20 +56,24 @@ const (
...
 	DeleteFilter(iface string, priority uint32) error
 	AddFwFilter(ifaces []string, parent string, handle string, flowid string) error
 	AddOutputLimit(ifaces []string, parent string, handle string, bytesPerSec uint) error
 	ClearQdisc(ifaces []string) error
 }
 
+//go:generate mockery --name=tcExecuter --filename=tc_executer_mock.go
+
 type tcExecuter interface {
-	Run(args ...string) (exitCode int, stdout string, stderr error)
```
Comment (luphaz) [inline network/tc.go:72]: [Variadic](https://vektra.github.io/mockery/notes/#variadic-arguments) arguments does not act as we expect

Using `unroll-variadic: False` make the generation of the mock failing, let's keep it simpler for now

**Review by ptnapoleon — APPROVED** [review]

**Pull request merged**